server.js

```diff
@@ -201,12 +200,18 @@ function getRoom(socket) {
  * Generate word list
  *
  * @param settings settings of the room
+ * @param hostDict dictionary from host
  * @return list of words
  */
-function generateWords(settings) {
+function generateWords(settings, hostDict) {
+    let dict = [];
     let words = [];
     let used = {};
-    let dict = dicts[settings.dictionaryId];
+    if (settings.wordset === "hostDictionary") {
+        dict = hostDict;
+    } else {
+        dict = dicts[settings.dictionaryId];
+    }
```

Comment:
```suggestion
    const dict = (settings.wordset === "hostDictionary") ? hostDict : dicts[settings.dictionaryId];
```
Но предварительно нужно удалить строчку
```javascript
    let dict = [];
```
чуть выше.

P.S. Можно заменить
```javascript
    let words = [];
    let used = {};
```
на
```javascript
    const words = [];
    const used = {};
```